client/task_runner.go

```diff
@@ -93,6 +93,12 @@ type TaskRunner struct {
 	// Must acquire persistLock when accessing
 	taskDirBuilt bool
 
+	// createdResources are all the resources created by the task driver
+	// across all attempts to start the task.
+	//
+	// Must acquire persistLock when accessing
```

Comment:
Sorry could have been more clear. You can add a createdResourceLock and SaveState can acquire it